# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmIonCompile.cpp
# Commit: 1ab8383d17ee
# Full Hash: 1ab8383d17ee1e5db164339236d64bcb7168b225
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 1
# Description:
#   Bug 1727084 - Memory64 - Atomic-rmw baseline operations. r=yury
#   
#   Atomic ADD, SUB, AND, OR, XOR, XCHG, CMPXCHG, as well as WAIT and
#   NOTIFY for the baseline compiler.
#   
# ==============================================================================

diff -r 6089b4360edc -r 1ab8383d17ee js/src/wasm/WasmIonCompile.cpp
--- a/js/src/wasm/WasmIonCompile.cpp	Mon Oct 18 07:31:55 2021 +0000
+++ b/js/src/wasm/WasmIonCompile.cpp	Mon Oct 18 07:31:56 2021 +0000
@@ -3556,7 +3556,7 @@
   uint32_t lineOrBytecode = f.readCallSiteLineOrBytecode();
 
   const SymbolicAddressSignature& callee =
-      type == ValType::I32 ? SASigWaitI32 : SASigWaitI64;
+      type == ValType::I32 ? SASigWaitI32M32 : SASigWaitI64M32;
   CallCompileState args;
   if (!f.passInstance(callee.argTypes[0], &args)) {
     return false;
@@ -3615,7 +3615,7 @@
 static bool EmitWake(FunctionCompiler& f) {
   uint32_t lineOrBytecode = f.readCallSiteLineOrBytecode();
 
-  const SymbolicAddressSignature& callee = SASigWake;
+  const SymbolicAddressSignature& callee = SASigWakeM32;
   CallCompileState args;
   if (!f.passInstance(callee.argTypes[0], &args)) {
     return false;